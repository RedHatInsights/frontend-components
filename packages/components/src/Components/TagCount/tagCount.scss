button.ins-c-tag-count {
    color: var(--pf-global--icon--Color--light);
    display: flex;
    align-content: center;
}

.ins-c-tag__text {
    margin-left: 10px;
}
