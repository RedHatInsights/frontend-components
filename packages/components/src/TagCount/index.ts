export * from './TagCount';
export { default } from './TagCount';
export { default as TagCount } from './TagCount';
