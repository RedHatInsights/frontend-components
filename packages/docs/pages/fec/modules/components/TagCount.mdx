
# TagCount

## Props

|name|type|default|description|
|----|----|-------|-----------|
|count|`number`|||
|onTagClick|React.MouseEventHandler&lt;HTMLButtonElement&gt;|`() => undefined`||
|className|`string`|||



