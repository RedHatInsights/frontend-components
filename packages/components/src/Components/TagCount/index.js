export { default as TagCount } from './TagCount';
import './tagCount.scss';
