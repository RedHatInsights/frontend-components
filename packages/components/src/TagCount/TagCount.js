import React from 'react';
import PropTypes from 'prop-types';
import './tagCount.scss';
import classNames from 'classnames';
import { Button } from '@patternfly/react-core';
import { TagIcon } from '@patternfly/react-icons';

const TagCount = ({ count, onTagClick, className, ...props }) => {
    return (
        <Button  {...props} variant="plain" isDisabled={!count} className={classNames('ins-c-tag-count', className)} onClick={onTagClick}>
            <TagIcon size="md"/>
            <span className="ins-c-tag__text">{count}</span>
        </Button>
    );
};

TagCount.propTypes = {
    count: PropTypes.number,
    onTagClick: PropTypes.func,
    className: PropTypes.string
};

TagCount.defaultProps = {
    onTagClick: () => undefined
};

export default TagCount;
