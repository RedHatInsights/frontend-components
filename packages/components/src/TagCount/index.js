export { default } from './TagCount';
export { default as TagCount } from './TagCount';
import './tagCount.scss';
