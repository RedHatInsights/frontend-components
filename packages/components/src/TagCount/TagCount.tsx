import React from 'react';
import TagCountPF from '@patternfly/react-component-groups/dist/dynamic/TagCount';
import { ButtonProps } from '@patternfly/react-core/dist/dynamic/components/Button';
export interface TagCountProps extends ButtonProps {
  count?: number;
  onTagClick?: React.MouseEventHandler<HTMLButtonElement>;
  className?: string;
}

/**
 * @deprecated Do not use deprecated TagCount import, the component has been moved to @patternfly/react-component-groups
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const TagCount: React.FunctionComponent<TagCountProps> = ({ onTagClick, ...props }) => <TagCountPF onClick={onTagClick} iconSize="md" {...props} />;

export default TagCount;
