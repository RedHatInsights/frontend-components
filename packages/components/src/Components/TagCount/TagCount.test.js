import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TagCount from './TagCount';

describe('TagCount component', () => {
    it('should render a disabled tag count with count 0', () => {
        const wrapper = shallow(<TagCount/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a tag count with no value', () => {
        const wrapper = shallow(<TagCount/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a tag count of 11', () => {
        const wrapper = shallow(<TagCount count="11"/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
