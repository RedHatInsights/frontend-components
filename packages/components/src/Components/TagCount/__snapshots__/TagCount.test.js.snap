// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCount component should render a disabled tag count with count 0 1`] = `
<Component
  className="ins-c-tag-count"
  isDisabled={true}
  onClick={[Function]}
  variant="plain"
>
  <TagIcon
    color="currentColor"
    noVerticalAlign={false}
    size="md"
    title={null}
  />
  <span
    className="ins-c-tag__text"
  />
</Component>
`;

exports[`TagCount component should render a tag count of 11 1`] = `
<Component
  className="ins-c-tag-count"
  isDisabled={false}
  onClick={[Function]}
  variant="plain"
>
  <TagIcon
    color="currentColor"
    noVerticalAlign={false}
    size="md"
    title={null}
  />
  <span
    className="ins-c-tag__text"
  >
    11
  </span>
</Component>
`;

exports[`TagCount component should render a tag count with no value 1`] = `
<Component
  className="ins-c-tag-count"
  isDisabled={true}
  onClick={[Function]}
  variant="plain"
>
  <TagIcon
    color="currentColor"
    noVerticalAlign={false}
    size="md"
    title={null}
  />
  <span
    className="ins-c-tag__text"
  />
</Component>
`;
